.github/workflows/im-reusable-setup-deployment-workflow.yml

```diff
@@ -0,0 +1,100 @@
+# The purpose of this workflow is to verify that the branch/tag/sha to deploy is a valid ref
+# and for production deploys to verify that the tag is reachable from the default branch
+# (meaning it has been reviewed and merged into main) and that the release is production
+# ready (it is not a draft/pre-release).
+
+
+# Example Usage in a repo's workflow:
+# jobs:
+#  setup-deployment-workflow:
+#    uses: im-practices/.github/.github/workflows/im-reusable-setup-deployment-workflow@v2
+#    with:
+#      ref-to-deploy: v1.2.3
+#      deployment-environment: prod
+#      production-environments: 'prod,prod-secondary'
+#      default-branch: main
+
+on:
+  workflow_call:
+    inputs:
+      ref-to-deploy:
+        description: The branch, tag or sha that will be deployed.
+        type: string
+        required: true
+      deployment-environment:
+        description: 'The environment that is being deployed to: dev, qa, stage, stage-secondary, uat, demo, prod, prod-secondary.'
+        required: true
+        type: string
+      production-environments:
+        description: Comma separated list of production environments to check against.  Defaults to 'prod,prod-secondary'
+        required: false
+        type: string
+        default: 'prod,prod-secondary'      
+      verify-release-production-ready:
+          description: Verify associated release is not draft or prerelease
+          required: false
+          type: boolean
+          default: true
+      default-branch:
+        description: Default branch of the repository
+        required: false
+        type: string
+        default: main
+      
+jobs:
+  setup-deployment-workflow:
+    runs-on: im-linux
+
+    steps:
+      - run: |
+          echo $'
+          | Deployment Arguments   |  Value                                              |
+          | ---                    |  ---                                                |
+          | Deployment Environment | `${{ inputs.deployment-environment }}`              |
+          | Ref to Deploy          | `${{ inputs.ref-to-deploy }}`                       |
+          | Actor                  | `${{ github.actor }}`                               |
+          | Workflow Source        | `${{ github.ref_name }}` - SHA: `${{ github.sha }}` |' >> $GITHUB_STEP_SUMMARY'
```

Comment:
Can this be toggled?  I know my team and others are recording their deployment arguments in a different way or in addition to and would like to customize this.

Reply:
I can update this so you can override the summary value.  I'll include that in the next PR and make sure that workflow also does it in a toggle-able way.